### PR TITLE
Add custom header support

### PR DESCRIPTION
**Pull request by isilher**
Hi! Thanks for making and maintaining this library. It's helped me add some very nice collapsing header behaviour to our react-navigation setup in the past days. I felt inspired by #109 and decided to give it a go. Let me know if this is of any use, I attempted to stay as close as possible to the current architecture and I'm happy to make some changes if required.

### Demo

demo-sample-3

### Example use

```js
function App() {
  return (
    <NavigationContainer
      /* Add headerMode="screen" to prevent the custom header from clashing with subsequent headers.
         If you don't do this, you will have to make sure the header is applied consistently.
         You can check the Custom Header implementation example to see a possible configuration for this */
      headerMode="screen"
    >
      <Stack.Navigator>
        /* Wrap your Stack.Screen */
        {createCollapsibleStack(
          <Stack.Screen
            name="HomeScreen"
            component={MyScreen}
            options={{
              title: 'Home',
            }}
          />,
          {
            /* Add a custom header to the createCollapsibleStack options the same way
               you would add it to the Stack.Screen options */
            header: ({ scene, previous, navigation }) => {
              const { options } = scene.descriptor;
              const title =
                options.headerTitle !== undefined
                  ? options.headerTitle
                  : options.title !== undefined
                  ? options.title
                  : scene.route.name;
              return (
                <MyHeader
                  title={title}
                  leftButton={
                    previous ? <MyBackButton onPress={navigation.goBack} /> : undefined
                  }
                  style={options.headerStyle}
                />
              );
            };
          }
        )}
      </Stack.Navigator>
    </NavigationContainer>
  );
}
```

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -27,4 +28,5 @@ export type CollapsibleStackConfig = {
   ) => ReactNode;
   key?: string;
   elevation?: number;
+  header?: ((props: StackHeaderProps) => React.ReactNode) | undefined;
```
Comment (benevbright) [inline src/types.ts:31]: you don't need ` | undefined` since it's `header?:`.

**Comment by isilher** [comment]
Thanks! I can update after the weekend. I'll have a look at the floating header mode too. Do you think it could be part of this change or better to tackle it separately?

**Comment by benevbright** [comment]
@isilher Thanks 👍 IMO, separately :) This PR is good as it is.

**Comment by isilher** [comment]
I've updated the types :-)

**Comment by benevbright** [comment]
Thanks. `v5.6.4` includes this PR.

**Comment by rickiesmooth** [comment]
it looks like this functionality doesnt work anymore with the 5.7.0 release, `options.headerStyle` seem to be ignored now?

**Pull request merged**